app/process/handler/consensus-message/consensus-message.go

```diff
@@ -87,7 +91,12 @@ func (cmh ConsensusMessageHandler) errorHandler(payload []byte) {
 func (cmh ConsensusMessageHandler) handleEthTxMessage(m *validatorproto.TopicEthTransactionMessage) error {
 	// TODO: verify authenticity of transaction hash
 
-	return cmh.scheduler.Cancel(m.TransactionId)
+	fromAddress, err := cmh.ethClient.GetTransactionFromAddress(common.HexToHash(m.EthTxHash))
```

Comment:
Will this return the TX even if it is still in the mempool?
What if the node is not synced fully and it returns that the tx is not present. Given the fast transactions in hedera, I think that this is plausible. Maybe we should consider setting the "from" in the proto file?